graphs/possible_bipartition.py

```diff
@@ -1,12 +1,143 @@
 # Can be used for BFS
 from collections import deque 
 
+group_1 = []
+group_2 = []
+
+def add_dog(graph, dog):
```

Comment:
I believe this function will return False if the dog has already been put in a group, which I'm unsure is your intention here.

The logic might be able to be updated to check if the dog has been put in a group before and then by checking all of its neighbors: if they have been put in a group, you will need to check whether the group is different than the one you're placing the new dog into. If they haven't been put in a group yet, you can put them in the opposite group and then do a check on all its neighbors (perhaps recursively). 

Overall, wonderful work!